### PR TITLE
restore information about truncated responses to sql api

### PR DESCRIPTION
**Pull request by techdocsmith**
Restores information about truncated responses for SQL HTTP API

The refactor of the SQL API removed some information about error handling and truncated responses:
https://github.com/apache/druid/pull/14711#issuecomment-1970076591

This PR restores that information. 

Open question on whether or not it applies to query from deep storage or not.

cc: @gianm 

This PR has:

- [x ] been self-reviewed.

### CODE REVIEW

File: docs/api-reference/sql-api.md [hunk 1]
```diff
@@ -53,11 +53,25 @@ The request body takes the following properties:
 * `query`: SQL query string.
 
 * `resultFormat`: String that indicates the format to return query results. Select one of the following formats:
-  * `object`: Returns a JSON array of JSON objects with the HTTP response header `Content-Type: application/json`.
-  * `array`: Returns a JSON array of JSON arrays with the HTTP response header `Content-Type: application/json`.
-  * `objectLines`: Returns newline-delimited JSON objects with a trailing blank line. Returns the HTTP response header `Content-Type: text/plain`.
-  * `arrayLines`: Returns newline-delimited JSON arrays with a trailing blank line. Returns the HTTP response header `Content-Type: text/plain`.
-  * `csv`: Returns a comma-separated values with one row per line and a trailing blank line. Returns the HTTP response header `Content-Type: text/csv`.
+  * `object`: Returns a JSON array of JSON objects with the HTTP response header `Content-Type: application/json`.  
+     Each object's field names match the columns returned by the SQL query, and are provided in the same order as the SQL query.
```
Comment (techdocsmith) [inline docs/api-reference/sql-api.md:57]:
```suggestion
     Object field names match the columns returned by the SQL query in the same order as the SQL query.
```

File: docs/api-reference/sql-api.md [hunk 2]
```diff
@@ -53,11 +53,25 @@ The request body takes the following properties:
...
+
+  * `array`: Returns a JSON array of JSON arrays with the HTTP response header `Content-Type: application/json`.  
+     Each inner array has elements matching the columns returned by the SQL query, in order.
+
+  * `objectLines`: Returns newline-delimited JSON objects with the HTTP response header `Content-Type: text/plain`.  
+     Newline separation can make it easier to parse the entire response set as a stream if you don't have a streaming JSON parser.
+     To make it possible to detect a truncated response, this format includes a trailer of one blank newline character.
```
Comment (techdocsmith) [inline docs/api-reference/sql-api.md:64]:
```suggestion
     This format includes a trailer of one blank newline character so you can detect a truncated response.
```

File: docs/api-reference/sql-api.md [hunk 3]
```diff
@@ -53,11 +53,25 @@ The request body takes the following properties:
...
+
+  * `arrayLines`: Returns newline-delimited JSON arrays with the HTTP response header `Content-Type: text/plain`.  
+     Newline separation can make it easier to parse the entire response set as a stream if you don't have a streaming JSON parser.
```
Comment (techdocsmith) [inline docs/api-reference/sql-api.md:67]:
```suggestion
     Newline separation facilitates parsing the entire response set as a stream if you don't have a streaming JSON parser.
```

File: docs/api-reference/sql-api.md [hunk 4]
```diff
@@ -53,11 +53,25 @@ The request body takes the following properties:
...
+     To make it possible to detect a truncated response, this format includes a trailer of one blank newline character.
```
Comment (techdocsmith) [inline docs/api-reference/sql-api.md:68]:
```suggestion
 This format includes a trailer of one blank newline character so you can detect a truncated response.
```

File: docs/api-reference/sql-api.md [hunk 5]
```diff
@@ -53,11 +53,25 @@ The request body takes the following properties:
...
+
+  * `csv`: Returns comma-separated values with one row per line. Sent with the HTTP response header `Content-Type: text/csv`.  
+     Druid uses double quotes to escape individual field values. For example a value with a comma returns `"A,B"`.
+     If the field value contains a double quote character, Druid escapes it with a second double quote character.
+     For example `foo"bar` becomes `foo""bar`.
+     To make it possible to detect a truncated response, this format includes a trailer of one blank newline character.
```
Comment (techdocsmith) [inline docs/api-reference/sql-api.md:74]:
```suggestion
      This format includes a trailer of one blank newline character so you can detect a truncated response.
```

**Review by 317brian — CHANGES_REQUESTED** [review]
Mostly LGTM with some suggestions for readability

File: docs/api-reference/sql-api.md [hunk 6]
```diff
@@ -53,11 +53,25 @@ The request body takes the following properties:
...
+  * `array`: Returns a JSON array of JSON arrays with the HTTP response header `Content-Type: application/json`.  
+     Each inner array has elements matching the columns returned by the SQL query, in order.
+
+  * `objectLines`: Returns newline-delimited JSON objects with the HTTP response header `Content-Type: text/plain`.  
+     Newline separation facilitates parsing the entire response set as a stream if you don't have a streaming JSON parser.
+     This format includes a trailer of one blank newline character so you can detect a truncated response.
+
+  * `arrayLines`: Returns newline-delimited JSON arrays with the HTTP response header `Content-Type: text/plain`.  
+     Newline separation facilitates parsing the entire response set as a stream if you don't have a streaming JSON parser.
+ This format includes a trailer of one blank newline character so you can detect a truncated response.
```
Comment (techdocsmith) [inline docs/api-reference/sql-api.md:68]:
```suggestion
 This format includes a single trailing newline character so you can detect a truncated response.
```

File: docs/api-reference/sql-api.md [hunk 7]
```diff
@@ -53,11 +53,25 @@ The request body takes the following properties:
...
+
+  * `csv`: Returns comma-separated values with one row per line. Sent with the HTTP response header `Content-Type: text/csv`.  
+     Druid uses double quotes to escape individual field values. For example a value with a comma returns `"A,B"`.
+     If the field value contains a double quote character, Druid escapes it with a second double quote character.
+     For example `foo"bar` becomes `foo""bar`.
+      This format includes a trailer of one blank newline character so you can detect a truncated response.
```
Comment (techdocsmith) [inline docs/api-reference/sql-api.md:74]:
```suggestion
      This format includes a single trailing newline character so you can detect a truncated response.
```

File: docs/api-reference/sql-api.md [hunk 8]
```diff
@@ -54,24 +54,24 @@ The request body takes the following properties:
...
-     Newline separation can make it easier to parse the entire response set as a stream if you don't have a streaming JSON parser.
-     To make it possible to detect a truncated response, this format includes a trailer of one blank newline character.
+     Newline separation facilitates parsing the entire response set as a stream if you don't have a streaming JSON parser.
+     This format includes a single trailing newline character so you can detect a truncated response.
 
   * `arrayLines`: Returns newline-delimited JSON arrays with the HTTP response header `Content-Type: text/plain`.  
-     Newline separation can make it easier to parse the entire response set as a stream if you don't have a streaming JSON parser.
-     To make it possible to detect a truncated response, this format includes a trailer of one blank newline character.
+     Newline separation facilitates parsing the entire response set as a stream if you don't have a streaming JSON parser.
+ This format includes a single trailing newline character so you can detect a truncated response.
```
Comment (317brian) [inline docs/api-reference/sql-api.md:68]:
```suggestion
     This format includes a single trailing newline character so you can detect a truncated response.
```

**Review by 317brian — APPROVED** [review]
The fixes LGTM besides the lost spaces on line 68.

**Pull request merged**